src/models/utils.py

```diff
@@ -27,31 +29,37 @@ def chunk_array(x: Union[torch.Tensor, np.ndarray],
     ----------
     An iterator returning tuples of batches (x, y)
     """
-    num_sections = max(1, x.shape[0] // batch_size)
-    if type(x) == np.ndarray:
+    if type(x) is not tuple:
+        x = (x, )
+    x = cast(Tuple[Union[torch.Tensor, np.ndarray]], x)
+
+    num_sections = max(1, x[0].shape[0] // batch_size)
+    if type(x[0]) == np.ndarray:
         return _chunk_ndarray(x, y, num_sections, shuffle)
     else:
         return _chunk_tensor(x, y, num_sections, shuffle)
 
 
-def _chunk_ndarray(x: np.ndarray, y: np.ndarray,
+def _chunk_ndarray(x: Tuple[np.ndarray, ...], y: np.ndarray,
                    num_sections: int,
-                   shuffle: bool) -> Iterable[Tuple[np.ndarray, np.ndarray]]:
+                   shuffle: bool) -> Iterable[Tuple[Tuple[np.ndarray, ...], np.ndarray]]:
 
-    split_x, split_y = np.array_split(x, num_sections), np.array_split(y, num_sections)
-    return_arrays = list(zip(split_x, split_y))
+    split_x = [np.array_split(x_section, num_sections) for x_section in x]
+    split_y = np.array_split(y, num_sections)
+    return_arrays = list(zip(*split_x, split_y))
 
     if shuffle:
         shuffle_list(return_arrays)
-    return return_arrays
+    return [(chunk[:-1], chunk[-1]) for chunk in return_arrays]  # type: ignore
```

Comment:
this returns separate chunks for the additional / final vector of `current` data?

Reply:
It returns a tuple for all the x data vectors, and the y data.

This allows you to iterate using:

```python
for x, y in dataloader:
    ...
```
where x is now a tuple of arrays / tensors